modules/nexverseBidAdapter.js

```diff
@@ -0,0 +1,447 @@
+  if (bidderRequest.uspConsent) {
+    openRtbRequest.regs.ext.us_privacy = bidderRequest.uspConsent;
+  }
+
+  return openRtbRequest;
+}
+
+/**
+ * Checks if the request is made over a secure connection (HTTPS).
+ * @returns {boolean} True if the connection is secure (HTTPS), false otherwise.
+ */
+function isSecureRequest() {
+  return location.protocol === 'https:';
+}
+
+/**
+ * Determines the device connection type.
+ * @returns {number} The connection type based on OpenRTB 2.5 spec.
+ */
+function getConnectionType() {
+  const connection = navigator.connection || navigator.mozConnection || navigator.webkitConnection;
+  if (connection) {
+    const type = connection.effectiveType || connection.type;
+    switch (type) {
+      case 'ethernet':
+        return 1; // Ethernet
+      case 'wifi':
+      case 'wimax':
+        return 2; // WiFi
+      case 'cellular':
+        return getCellularConnectionType(connection);
+      default:
+        return 0; // Unknown
+    }
+  }
+  return 0; // Unknown
+}
+
+/**
+ * Determines the cellular connection type based on OpenRTB 2.5 spec.
+ * @param {Object} connection - The connection object from the browser.
+ * @returns {number} The cellular connection type.
+ */
+function getCellularConnectionType(connection) {
+  const gen = connection.effectiveType;
+  switch (gen) {
+    case '2g':
+      return 4; // Cellular (2G)
+    case '3g':
+      return 5; // Cellular (3G)
+    case '4g':
+      return 6; // Cellular (4G)
+    case '5g':
+      return 7; // Cellular (5G)
+    default:
+      return 3; // Cellular (Unknown Generation)
+  }
+}
+
+/**
+ * Determines the device type (1 = Mobile/Tablet, 2 = Desktop).
+ * @returns {number} The device type.
+ */
+function getDeviceType() {
+  const ua = navigator.userAgent;
+  if (/Mobi|Android|iPhone|iPad/i.test(ua)) {
+    return 1; // Mobile/Tablet
+  }
+  return 2; // Desktop
+}
+
+/**
+ * Determines the device operating system.
+ * @returns {string} The OS name.
+ */
+function getDeviceOS() {
+  const ua = navigator.userAgent;
+  if (/Android/i.test(ua)) {
+    return 'Android';
+  } else if (/iPhone|iPad|iPod/i.test(ua)) {
+    return 'iOS';
+  } else if (/Windows/i.test(ua)) {
+    return 'Windows';
+  } else if (/Mac OS/i.test(ua)) {
+    return 'Mac OS';
+  }
+  return 'Other';
+}
+
+/**
+ * Determines the device operating system version.
+ * @returns {string} The OS version.
+ */
+function getDeviceOSVersion() {
+  const ua = navigator.userAgent;
+  let osVersion = 'unknown';
+
+  if (/Android/i.test(ua)) {
+    const match = ua.match(/Android\s([0-9\.]+)/);
+    if (match) {
+      osVersion = match[1];
+    }
+  } else if (/iPhone|iPad|iPod/i.test(ua)) {
+    const match = ua.match(/OS\s([0-9_]+)/);
+    if (match) {
+      osVersion = match[1].replace(/_/g, '.');
+    }
+  }
+  return osVersion;
+}
+
+/**
+ * Determines the device model (if possible).
+ * @returns {string} The device model.
+ */
```

Comment:
Please get this from the request, if the request is unsatisfactory, please modify it in core

Reply:
Fixed in new PR https://github.com/prebid/Prebid.js/pull/12472

Reply:
Done.